spec/models/king_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe King, type: :model do
+  describe '#create' do
+    it 'creates a king in the database' do
+      king = FactoryGirl.create(:king)
+      expect(king.type == 'King').to eq(true)
+    end
+  end
+
+  describe '#valid_move?' do
+    it 'returns true if the move is diagonal up-right one square' do
+      king = FactoryGirl.create(:king, x_position: 4, y_position: 4)

```

Comment:
OK, thanks Jack. Just did `git add --all` and `git commit -am "comment"`, both of which worked as expected. Then tried `git push origin add_king_logic` and got this response:
> ! [rejected]        add_king_logic -> add_king_logic (fetch first)
> error: failed to push some refs to 'git@github.com:firehose-codemasters/codemasters_chess.git'
> hint: Updates were rejected because the remote contains work that you do
> hint: not have locally. This is usually caused by another repository pushing
> hint: to the same ref. You may want to first integrate the remote changes
> hint: (e.g., 'git pull ...') before pushing again.
> hint: See the 'Note about fast-forwards' in 'git push --help' for details.